File: src/NzbDrone.Core/Profiles/Tagging/TaggingProfile.cs
using System.Collections.Generic;
using NzbDrone.Core.Configuration;
using NzbDrone.Core.Datastore;

namespace NzbDrone.Core.Profiles.Tagging
{
    public class TaggingProfile : ModelBase
    {
        public string Name { get; set; }
        public HashSet<int> Tags { get; set; }
        public int Order { get; set; }
        public WriteAudioTagsType WriteAudioTags { get; set; }
        public bool ScrubAudioTags { get; set; }
        public bool EmbedCoverArt { get; set; }
        public List<int> DownloadClientIds { get; set; }
        public bool SkipHardlinkedFiles { get; set; }

        public TaggingProfile()
        {
            Tags = new HashSet<int>();
            DownloadClientIds = new List<int>();
            EmbedCoverArt = true;
        }
    }
}

File: frontend/src/typings/TaggingProfile.ts
import ModelBase from 'App/ModelBase';

export type WriteAudioTagsType = 'no' | 'newFiles' | 'allFiles' | 'sync';

interface TaggingProfile extends ModelBase {
  name: string;
  writeAudioTags: WriteAudioTagsType;
  scrubAudioTags: boolean;
  embedCoverArt: boolean;
  order: number;
  tags: number[];
  skipHardlinkedFiles: boolean;
}

export default TaggingProfile;
